pfcpiface/conn.go

```diff
@@ -62,6 +62,12 @@ func pfcpifaceMainLoop(upf *upf, accessIP, coreIP, sourceIP, smfName string) {
 		return
 	}
 
+	ips, err := net.LookupIP("upf")
+	if err == nil {
+		sourceIP = ips[0].String()
+		log.Println("Service IP : ", sourceIP)
+	}
+
```

Comment:
Lets not hardcode this. Also any figuring out of sourceIP is done in main itself. Please modify the cpiface block itself. There should only 2 optional IP/Host related configs in case current names are unclear.
1. hostLocal
2. hostRemote

User can give name or IP. There can be 4 cases
a. Both are specified - Use as is and initiate the connection
b. hostLocal only is specified - Bind to local and wait for a connection
c. hostRemote only is specified - Figure out the local IP that lets you reach the remote. We have code for this already
d. Neither is specified - bind to any and wait for a connection

Reply:
this change is done to send service IP to the SMF/SPGW instead of node IP. Initially we use nodeIP as the sourceIP to bind locally.. we cannot bind to service IP. so this is needed. After that's done we use the same IP to be sent as nodeIP in all messages. So updated it to use Service IP. I can update it to be a different variable called Service IP and figure it out at main itself. is that ok ?

Reply:
This is leaking k8s details into the code which we don't want. Can you tell what is the aim of introducing service IP into the mix? I did not quite understand. Is there flow diagram/doc I can take a look at?

Reply:
the pods can restart.. so pod ip is not consistent.. if we give service ip.. then smf can use that as upf's node id.. 
<img width="390" alt="Screen Shot 2021-03-18 at 5 08 21 PM" src="https://user-images.githubusercontent.com/8911083/111713341-ab4f0600-880c-11eb-94f8-311d1fe430c6.png">

Reply:
currently `fqdnHost` seems unused. Can we use this? Also the pods in the statefulset get a stable DNS name across re-spins so that should be enough instead of extra serviceIP. wdyt?